### PR TITLE
Display actual keybindings in the UI by reading the keybinding.json

### PR DESCRIPTION
**Pull request by hassoncs**
WIP

This commit introduces a new utility for reading and displaying platform-specific keybindings within the VS Code extension since VSC does not provide an API for it! 😭 

Key changes include:
- **New `keybindings.ts` utility**: Provides functions to retrieve current keybinding labels for commands, considering user overrides and default keybindings. It also handles platform-specific formatting (e.g., Cmd on macOS, Ctrl on Windows/Linux) and chord keybindings.
- **New `vscode-config.ts` utility**: Handles reading user `keybindings.json` files, supporting various VS Code forks (VSCodium, Cursor, Windsurf) and portable installations. It parses JSON5 to allow comments and trailing commas.
- **Webview integration**: The `GhostServiceSettings` component now uses a new `useKeybindings` hook to dynamically fetch and display the correct keybinding for the "Quick Inline Task" in its description. This ensures users see the shortcut relevant to their OS and custom settings.
- **Terminal Welcome Service update**: The `TerminalWelcomeService` now uses the new keybinding utility to display the correct shortcut for generating terminal commands.
- **Message passing**: Added new `getKeybindings` and `keybindings` message types for communication between the webview and the extension backend to facilitate keybinding retrieval.
- **Internationalization updates**: Modified i18n files across multiple languages to support dynamic keybinding interpolation in relevant strings.
- **Tests**: Comprehensive unit and integration tests for the new keybinding and vscode-config utilities, covering various platforms, file formats, and edge cases.

This feature enhances user experience by providing accurate and context-aware keybinding information directly within the UI, reducing confusion and improving discoverability of shortcuts.## Context

## Screenshots

| before | after |
| ------ | ----- |
|        |       |

### CODE REVIEW

File: src/package.json [hunk 1]
```diff
@@ -715,6 +716,7 @@
 		"@types/diff": "^5.2.3",
 		"@types/diff-match-patch": "^1.0.36",
 		"@types/glob": "^8.1.0",
+		"@types/json5": "^2.2.0",
```
Comment (markijbema) [inline src/package.json:719]: why does this need an extra json library?

File: src/utils/keybindings.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+ * Gets the default keybinding for a command from package.json
+ * @param commandId The command ID to look up
+ * @returns The platform-specific keybinding string (guaranteed to exist for valid commands)
+ * @throws Error if the command is not found in package.json keybindings
+ */
+export function getDefaultKeybindingForCommand(commandId: string): string {
+	const keybindings = packageJson.contributes?.keybindings as KeybindingEntry[] | undefined
+	const binding = keybindings?.find((kb) => kb.command === commandId)
+	if (!binding) {
+		throw new Error(`Command '${commandId}' not found in package.json keybindings`)
```
Comment (markijbema) [inline src/utils/keybindings.ts:99]: given the purpose of this functionality (showing the user a key), I would say return null or 'not found' would be better. If we want to detect it we can still send an event to posthog, but it doesnt seem important enough to throw na exception right?

File: src/utils/keybindings.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+function prettyPrintKey(rawKey: string, platform: NodeJS.Platform): string {
+	const chords = rawKey.split(" ").filter(Boolean)
+
+	const formattedChords = chords.map((chord) => {
+		const parts = chord.split("+")
+		return parts.map((part) => normalizeKeyToken(part, platform)).join("+")
+	})
+
+	if (formattedChords.length > 1) {
+		return formattedChords.join(", ")
```
Comment (markijbema) [inline src/utils/keybindings.ts]: join works fine on an array of one or zero elements as well right?

File: src/shared/WebviewMessage.ts [hunk 4]
```diff
@@ -335,6 +336,7 @@ export interface WebviewMessage {
 	filters?: { type?: string; search?: string; tags?: string[] }
 	settings?: any
 	url?: string // For openExternal
+	commandIds?: string[] // kilocode_change: For getKeybindings
```
Comment (markijbema) [inline src/shared/WebviewMessage.ts]: for merging its easier to keep all kilocode_changes next to eachother

**Review by markijbema — APPROVED** [review]
Because this is mostly 'view' and not actually invoked in any logic i think it is fine. But I think it will be better to have it be robust and not crash, because the use also doesnt really is worth to crash for

File: src/package.json (hunk 1 above)
Comment (hassoncs) [reply src/package.json:719]: The vscode json files are stored in JSON5 format which is just JSON that allows comments. I thought about just implementing it ourselves, but I'd rather use a tested library for it, as silly as it is to need it just for this

File: src/utils/keybindings.ts (hunk 2 above)
Comment (hassoncs) [reply src/utils/keybindings.ts:99]: Yeah! I agree- I'll update it to show "not found" in the UI if there's no binding found 👍

File: src/utils/keybindings.ts (hunk 3 above)
Comment (hassoncs) [reply src/utils/keybindings.ts]: Definitely, vibed too hard here :)
